src/OMSimulatorLib/BusConnector.h

```diff
@@ -0,0 +1,51 @@
+#ifndef BUSCONNECTOR_H
+#define BUSCONNECTOR_H
+
+
+#include "ComRef.h"
+#include "SignalRef.h"
+#include "Types.h"
+#include "Connector.h"
+#include "ssd/ConnectorGeometry.h"
+
+#include <string>
+#include <vector>
+
+#include <pugixml.hpp>
+
+namespace oms3
+{
+  /**
+   * \brief Connector
+   */
+  class BusConnector : protected oms_connector_t
+  {
+  public:
+    /**
+     * This constructor creates a oms2::Connector without geometry information.
+     */
+    BusConnector(const oms3::ComRef& name);
+    ~BusConnector();
+
+    oms_status_enu_t exportToSSD(pugi::xml_node& root) const;
+
+    // methods to copy the object
+    BusConnector(const BusConnector& rhs);
+    BusConnector& operator=(const BusConnector& rhs);
+
+    void setName(const oms3::ComRef& name);
+    void setGeometry(const oms2::ssd::ConnectorGeometry* newGeometry);
+
+    const oms_causality_enu_t getCausality() const {return causality;}
+    const oms_signal_type_enu_t getType() const {return type;}
+    const oms3::ComRef getName() const {return oms3::ComRef(std::string(name));}
+    const oms2::ssd::ConnectorGeometry* getGeometry() const {return reinterpret_cast<oms2::ssd::ConnectorGeometry*>(geometry);}
+
+    oms_status_enu_t addConnector(const oms3::ComRef& cref);
+
+  private:
+    std::vector<oms3::ComRef> connectors;
```

Comment:
I would recommend to introduce a new type for the C interface. Using `oms_connector_t` is a bit problematic, because it looks different in memory compared to `BusConnector`.

Reply:
Good point. A bus connector is different from a regular connector. I added a new struct called oms3_busconnector_t.